### DISCUSSION TITLE
Adding test name to the pytest_runtest_call

### DISCUSSION THREAD

**Discussion by f00dog-x300** [pytest-dev/pytest#9376] [Q&A] [answered]
Hello everyone, 

I'm trying to add to the log the name of the test case that was being run when the test data is being generated. For experimentation purposes, I tried to add the following to the `conftest.py` file: 
```
import logging

LOGGER = logging.getLogger(__name__)

def pytest_runtest_call(item, request):
    LOGGER.info(f"running before each test: {request.node.name}")
    
def pytest_runtest_teardown(item, request):
    LOGGER.info(f"running after test: {request.node.name}")
```

The error message I'm getting is: 
`Argument(s) {'request'} are declared in the hookimpl but can not be found in the hookspec`

which seems to suggest that I'm not supposed to be using it this way. 

My question is if there is a way for me to grab the test case name somehow before each test is run. The other way I could think of is going to each test file and trying to create a test fixture that would grab it that way, but thinking that if there are a lot of test files, that's going to be a bit of a pain to implement.

**Comment by The-Compiler** [comment, marked as answer]
[`request`](https://docs.pytest.org/en/6.2.x/reference.html#std-fixture-request) is a pytest fixture, i.e. exposed via a mechanism you can only access from tests or from other fixtures, but not from hooks (which, you could say, are more low-level in a sense).

However, the `item` argument of [`pytest_runtest_call`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.hookspec.pytest_runtest_call) is the test [`Item`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.Item) - and `Item` is a subclass of [`Node`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.nodes.Node), which is what [`request.node`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.FixtureRequest.node) points at (*Underlying collection node (depends on current request scope).*).

Long story short: `item.name` should get you there!

**Reply by f00dog-x300** [reply]
I appreciate it! This is what I was looking for. Is there a way to pass another fixture inside these methods?

**Reply by The-Compiler** [reply]
No, fixtures only work with tests, not with hooks.
